src/cpu.rs

```diff
@@ -5565,5 +5634,346 @@ mod test {
     }
 
 	// --------- ROL ---------
+
+	// test cases to fulfill for ROL:
+
+	// happy path
+    	// 0x08 = 0000 1000  ->  0001 0000 = 0x10
+
+	// 0 with carry bit set
+    	// 0x00 = 0000 0000  ->  0000 0000 = 0x00 -> 0000 0001 = 0x01
+
+	// 0 without carry bit set
+    	// 0x00 = 0000 0000  ->  0000 0000 = 0x00
+
+    // negative value with carry set
+    	// 0x80 = 1000 0000  ->  0000 0000 = 0x00 -> 0000 0001 = 0x01
+
+	// negative value without carry set
+    	// 0x80 = 1000 0000  ->  0000 0000 = 0x00
+
+    // all middle bits set, end bits not set, no carry bit set
+    	// 0x7e = 0111 1110  ->  1111 1100 = 0xfc
+
+	// all middle bits set, end bits not set, carry bit set
+    	// 0x7e = 0111 1110  ->  1111 1100 = 0xfc -> 1111 1101 = 0xfd
+
+	// ------- accumulator --------
+    #[test]
+	fn test_rol_accumulator_happy_path() {
+		// create a cpu
+		let mut cpu = CPU::new();
+
+		// Load and run a short program.
+		// 1. Load a positive value into A, it wouldn't become negative after shift.
+    	// 2. Perform the rotate left on the accumulator value.
+    	// 3. Break.
+    	cpu.load_and_run(vec![0xa9, 0x08, 0x2a, 0x00]);
+
+    	// Check that the a value is expected.
+    	// 0x08 = 0000 1000  ->  0001 0000 = 0x10
+    	assert_eq!(cpu.a, 0x10);
+
+    	// Check that the p register is expected.
+    	assert_eq!(cpu.p, 0b0000_0000);
+	}
+
+    #[test]
+	fn test_rol_accumulator_zero_carry_set() {
+		// create a cpu
+		let mut cpu = CPU::new();
+
+		// Load and run a short program.
+		// 1. Load zero into A, it wouldn't become negative after shift.
+		// 2. Put a value in memory.
```

Comment:
[REMINDER] be consistent. use either:

* Put the a value into memory
* Store the a value into memory
* Load a value into memory